tests/test_transforms.py

```diff
@@ -258,3 +262,24 @@ def transform_class(cls):
                 import UserDict
         """
         )
+
+    def test_transform_aborted_if_recursion_limited(self):
+        def transform_call(node: Call) -> Const:
+            return node
+
+        self.transformer.register_transform(
+            nodes.Call, transform_call, _looks_like_dataclass_field_call
+        )
+
+        if IS_PYPY:
+            original_limit = 1000  # pypy doesn't expose this
+            sys.setrecursionlimit(600)
+        else:
+            original_limit = sys.getrecursionlimit()
+            sys.setrecursionlimit(1000)  # use the default
+
+        try:
+            with self.assertWarns(UserWarning):
```

Comment:
Might want to assert on the message as well to make sure we're not catching something else?